xla/service/gpu/nccl_api.h

```diff
@@ -226,8 +241,19 @@ class NcclApi {
   // https://docs.nvidia.com/deeplearning/nccl/user-guide/docs/api/comms.html#ncclcommderegister
   virtual absl::StatusOr<NcclRegisteredBufferHandle> DeregisterBuffer(
       NcclCommHandle comm, NcclRegisteredBufferHandle handle) = 0;
-};
 
+  // Returns if initializing communicator with custom configuration is supported
+  // or not. Available after 2.17.0.
+  // https://docs.nvidia.com/deeplearning/nccl/user-guide/docs/api/comms.html#ncclcomminitrankconfig
+  virtual absl::StatusOr<bool> IsInitConfigSupported() = 0;
+
+  // Creates a new communicator with custom configuration.
+  //
+  // https://docs.nvidia.com/deeplearning/nccl/user-guide/docs/api/comms.html#ncclcomminitrankconfig
+  virtual absl::StatusOr<OwnedNcclComm> CommInitRankConfig(
+      int32_t nranks, const NcclCliqueId& clique_id,
+      int32_t rank, NcclCommConfig& config) = 0;
```

Comment:
instead of adding new api just pass config in std::optional with default nullopt